src/core/Akka.Persistence.TestKit/PersistenceTestKit.cs

```diff
@@ -0,0 +1,64 @@
+//-----------------------------------------------------------------------
+// <copyright file="PersistenceTestKit.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2019 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2019 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Akka.Persistence.TestKit
+{
+    using Actor;
+    using Akka.TestKit;
+    using Akka.TestKit.Xunit;
+    using Configuration;
+    using System;
+
+    public abstract class PersistenceTestKit : TestKitBase
+    {
+        protected PersistenceTestKit(string actorSystemName = null, string testActorName = null)
+            : base(new XunitAssertions(), GetConfig(), actorSystemName, testActorName)
+        {
+            JournalActorRef = GetJournalRef(Sys);
+            Journal = TestJournal.FromRef(JournalActorRef);
+        }
+
+        public IActorRef JournalActorRef { get; }
+
+        public ITestJournal Journal { get; }
+
+        public void WithFailingJournalRecovery(Action execution)
```

Comment:
@valdisz here's an example of this method on the implementation side -we should expose some more of these for both the TestJournal and the TestSnapshotstore. We can implement the latter in a separate pull request - but does this make sense?

Reply:
This is clear, I agree that it will be cleaner.

Reply:
End-user always must do their assertions inside `execution` callback. Otherwise, it could lead to unexpected results when Journal behavior is reverted, but actor under test have not to even tried to persist anything into Journal.

Reply:
@Aaronontheweb how about such API?

``` csharp
public async Task WithJournalRecovery(Func<JournalRecoveryBehavior, Task> behaviorSelector, Func<Task> execution)
{
    try
    {
        await behaviorSelector(Journal.OnRecovery);
        await execution();
     }
    finally
    {
        await Journal.OnRecovery.Pass();
    }
}
```

and use it like this:

``` csharp
[Fact]
public async Task new_api()
{
    await WithJournalRecovery(recovery => recovery.Fail(), () =>
    {
        // code to trigger actor under test
        // and assertions
        return Task.CompletedTask;
    });
}
```

Reply:
> End-user always must do their assertions inside execution callback. Otherwise, it could lead to unexpected results when Journal behavior is reverted, but actor under test have not to even tried to persist anything into Journal.

Yep, I agree - that's what we should teach them to do in the documentation.

Reply:
I like the API design too - if our goal is to avoid a blocking testkit going forward, returning `Task` is the way to go.